great_expectations/experimental/datasources/interfaces.py

```diff
@@ -424,20 +435,33 @@ def get_asset(self, asset_name: str) -> _DataAssetT:
                 f"'{asset_name}' not found. Available assets are {list(self.assets.keys())}"
             ) from exc
 
-    def add_asset(self, asset: _DataAssetT) -> _DataAssetT:
+    def add_asset(
+        self,
+        asset: _DataAssetT,
+        data_connector: Optional[DataConnector] = None,
```

Comment:
Where is the code that actually creates the `DataConnector` instance?

Reply:
@Kilo59 It is in our `Datasource` subclasses, which are the ones that know exactly which `DataConnector` variant they should instantiate and have all the lowest-level arguments to construct them.